ext/php8/serializer.c

```diff
@@ -35,10 +42,22 @@ static int write_hash_table(mpack_writer_t *writer, HashTable *ht TSRMLS_DC) {
                 mpack_start_array(writer, zend_hash_num_elements(ht));
             }
         }
+
+        // Writing the key, if associative
+        bool zval_string_as_uint64 = false;
         if (is_assoc == 1) {
-            mpack_write_cstr(writer, ZSTR_VAL(string_key));
+            char *key = ZSTR_VAL(string_key);
+            mpack_write_cstr(writer, key);
+            // If the key is trace_id, span_id or parent_id then strings have to be converted to uint64 when packed.
+            if (0 == strcmp(KEY_TRACE_ID, key) || 0 == strcmp(KEY_SPAN_ID, key) || 0 == strcmp(KEY_PARENT_ID, key)) {
+                zval_string_as_uint64 = true;
+            }
         }
-        if (msgpack_write_zval(writer, tmp TSRMLS_CC) != 1) {
+
+        // Writing the value
+        if (zval_string_as_uint64) {
+            mpack_write_u64(writer, strtoull(Z_STRVAL_P(tmp), NULL, 10));
+        } else if (msgpack_write_zval(writer, tmp TSRMLS_CC) != 1) {
```

Comment:
Adding new TSRMLS - maybe rebase it, then compilation would be (remove it in PHP 7 and PHP 8)

Reply:
👍🏻  just pulled in master